Sami/Renderer/Renderer.php

```diff
@@ -227,12 +227,20 @@ protected function getProgression()
         return floor((++$this->step / $this->steps) * 100);
     }
 
+    /**
+     * Get tree for the given project.
+     *
+     * @param Project $project
+     *
+     * @return array
+     */
     private function getTree(Project $project)
     {
-        if (!isset($this->cachedTree[$project])) {
-            $this->cachedTree[$project] = $this->tree->getTree($project);
+        $key = $project->getVersion()->getName();
```

Comment:
The key should probably contain the project as well, no?

Reply:
That depends on average location the cache folder is placed. In default configuration cache folder for each of projects is different, so this shouldn't be an issue.

Reply:
If we are re-using a single renderer across multiple projects, then there is a risk that project a / version 1 could conflict with version b / version 1.

Would it be sufficient for me to use `$project->getBuildDir()` as the cache instead? This will include the `%version%` substitution, thus making it distinct between versions.

Reply:
Please see my latest commit which addresses this. :)